src/expr/src/relation/func.rs

```diff
@@ -513,6 +513,67 @@ impl AggregateFunc {
         };
         scalar_type.nullable(nullable)
     }
+
+    /// get the function that combines the results of partially
+    /// pushing down the aggregation + the function, if necessary, that
+    /// downcasts the type of the aggregation column back to the original
+    pub fn outer_agg(&self) -> (AggregateFunc, Option<crate::UnaryFunc>) {
+        match self {
+            AggregateFunc::Count => (
+                AggregateFunc::SumInt64,
+                Some(crate::UnaryFunc::CastDecimalToInt64(0)),
+            ),
+            AggregateFunc::SumInt32 => (
+                AggregateFunc::SumInt64,
+                Some(crate::UnaryFunc::CastDecimalToInt64(0)),
+            ),
+            AggregateFunc::SumInt64 => (AggregateFunc::SumDecimal, None),
+            _ => (self.clone(), None),
+        }
```

Comment:
This doesn't seem correct for things like `json_agg`.

Reply:
It is not correct for `jsonb_agg` but we don't have a function at the moment that flat-maps `jsonb` lists into one big list, and for that reason, reduction pushdown does not push down `jsonb_agg`. To make it clear that `jsonb_agg` cannot be combined using itself, I could have a separate branch here 
```
// we don't yet have an aggregation to flat map jsonb lists into bigger lists
AggregateFunc::JsonbAgg => unreachable!(),
```

Reply:
Actually, I've opted for making the function return an Option, and if an aggregation is like `JsonbAgg`, then it returns None. Any function for which outer_agg() returns None does not get partially pushed down.